src/python_testing/TestChoiceConformanceSupport.py

```diff
@@ -0,0 +1,211 @@
+#
+#    Copyright (c) 2023 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+import itertools
+import xml.etree.ElementTree as ElementTree
+
+import jinja2
+from choice_conformance_support import (evaluate_attribute_choice_conformance, evaluate_command_choice_conformance,
+                                        evaluate_feature_choice_conformance)
+from matter_testing_support import MatterBaseTest, ProblemNotice, default_matter_test_main
+from mobly import asserts
+from spec_parsing_support import XmlCluster, add_cluster_data_from_xml
+
+FEATURE_TEMPLATE = (
+    '    <feature bit="{{ id }}" code="{{ name }}" name="{{ name }}" summary="summary">\n'
+    '      <optionalConform choice="{{ choice }}" more="{{ more }}">\n'
+    '    {% if XXX %}'
+    '        <feature name="XXX" />\n'
+    '    {% endif %}'
+    '    </optionalConform>\n'
+    '    </feature>\n'
+)
```

Comment:
nit: did you consider multiline strings here?

```suggestion
FEATURE_TEMPLATE = '''\
    <feature bit="{{ id }}" code="{{ name }}" name="{{ name }}" summary="summary">
      <optionalConform choice="{{ choice }}" more="{{ more }}">
      {%- if XXX %}'
       <feature name="XXX" />
      {% endif %}
      </optionalConform>
    </feature>
'''
```